src/test/atlas_connectivity.rs

```diff
@@ -65,9 +65,78 @@ async fn atlas_serverless() {
 #[cfg_attr(feature = "tokio-runtime", tokio::test)]
 #[cfg_attr(feature = "async-std-runtime", async_std::test)]
 async fn atlas_serverless_srv() {
+    run_test("MONGO_ATLAS_SERVERLESS_URI_SRV", None).await;
```

Comment:
for the other existing SRV test above for free tier, we test both with cloudflare and with no resolver config specified. I used the same pattern for the new tests below so I figured I should add a corresponding check to this test.

Reply:
Given that serverless and atlas free tier are a bit different on the cloud side of things, I think I lean towards more coverage here, especially since almost all users will be using the default resolver config.